test/cpp/tensorexpr/test_simplify.cpp

```diff
@@ -2609,6 +2734,164 @@ void testSimplifyDivisionScalarFactorization() {
   }
 }
 
+void testSimplifyDivisionSymbolicFactorization() {
+  KernelScope kernel_scope;
+
+  {
+    // Same numerator and denomenator still cancel.
+    // x / x => 1.
```

Comment:
I think that transformation is not correct if `x` can be 0.

Reply:
This should be guarded by the check on line 1072.

Reply:
Oh I see what you mean, we don't know at simplification time if it will hit the `0 % x` branch or `x % x` branch. Good catch. We could replace it with `CompareSelect(x, 0, kEq, 0, 1)`? Is that better?

Reply:
Probably not right. Unfortunate, I guess remove this rule.